go/datas/database_common.go

```diff
@@ -27,84 +27,89 @@ var (
 )
 
 func newDatabaseCommon(cch *cachingChunkHaver, vs *types.ValueStore, rt chunks.RootTracker) databaseCommon {
-	return databaseCommon{ValueStore: vs, cch: cch, rt: rt, rootRef: rt.Root()}
+	return databaseCommon{ValueStore: vs, cch: cch, rt: rt, rootHash: rt.Root()}
 }
 
-func (ds *databaseCommon) MaybeHead(datasetID string) (types.Struct, bool) {
-	if r, ok := ds.MaybeHeadRef(datasetID); ok {
-		return r.TargetValue(ds).(types.Struct), true
-	}
-	return types.Struct{}, false
-}
-
-func (ds *databaseCommon) MaybeHeadRef(datasetID string) (types.Ref, bool) {
-	if r, ok := ds.Datasets().MaybeGet(types.String(datasetID)); ok {
+func (dbc *databaseCommon) maybeHeadRef(datasetID string) (types.Ref, bool) {
+	if r, ok := dbc.Datasets().MaybeGet(types.String(datasetID)); ok {
 		return r.(types.Ref), true
 	}
 	return types.Ref{}, false
 }
 
-func (ds *databaseCommon) Head(datasetID string) types.Struct {
-	c, ok := ds.MaybeHead(datasetID)
-	d.PanicIfFalse(ok, "Database \"%s\" has no Head.", datasetID)
-	return c
-}
-
-func (ds *databaseCommon) HeadRef(datasetID string) types.Ref {
-	r, ok := ds.MaybeHeadRef(datasetID)
-	d.PanicIfFalse(ok, "Database \"%s\" has no Head.", datasetID)
-	return r
-}
-
-func (ds *databaseCommon) Datasets() types.Map {
-	if ds.datasets == nil {
-		if ds.rootRef.IsEmpty() {
+func (dbc *databaseCommon) Datasets() types.Map {
```

Comment:
I'm surprised there isn't.

In this case, though, this type isn't exported, so the comment wouldn't show up in godoc